Casks/b/beeper.rb

```diff
@@ -1,39 +1,31 @@
 cask "beeper" do
-  arch arm: "arm64", intel: "x64"
+  arch arm: "-arm64"
+  livecheck_arch = on_arch_conditional arm: "arm64", intel: "x64"
 
-  version "3.110.1,241031sx3gh9lyl"
-  sha256 arm:   "632e61ba8ab9a9648cfcace1cfa89f53ad184398f647b70ff777ee0576a834d5",
-         intel: "2735ca5517b34be992424e2a45dfe448184bf40bd987f45499db09610ac3a7f6"
+  version "4.0.604"
+  sha256 arm:   "1c2bf883e87cd66340db6adf5d9e180f26e12507183d65f1d738be1249bf2fa3",
+         intel: "f62d9d75ceb7024ea189e86102dc67bd0b4b6b0eede0de3260db41e1362e5e4c"
 
-  url "https://download.todesktop.com/2003241lzgn20jd/Beeper%20#{version.csv.first}%20-%20Build%20#{version.csv.second}-#{arch}-mac.zip",
-      verified: "download.todesktop.com/2003241lzgn20jd/"
+  url "https://beeper-desktop.download.beeper.com/builds/Beeper-#{version}#{arch}-mac.zip"
   name "Beeper"
   desc "Universal chat app powered by Matrix"
   homepage "https://www.beeper.com/"
 
   livecheck do
-    url "https://download.todesktop.com/2003241lzgn20jd/latest-mac.yml"
-    regex(/Beeper\sv?(\d+(?:\.\d+)+)(?:\s-\sBuild\s([a-z\d]+?))?-#{arch}-mac\.zip/i)
-    strategy :electron_builder do |yaml, regex|
-      yaml["files"]&.map do |item|
-        match = item["url"]&.match(regex)
-        next if match.blank?
-
-        match[2].present? ? "#{match[1]},#{match[2]}" : match[1]
-      end
-    end
+    url "https://api.beeper.com/desktop/download/macos/#{livecheck_arch}/stable/com.automattic.beeper.desktop"
+    strategy :header_match
```

Comment:
```suggestion
    url "https://api.beeper.com/desktop/update-feed.json?bundleID=com.automattic.beeper.desktop&platform=macos&arch=#{livecheck_arch}&channel=stable"
    strategy :json do |json|
      json["version"]
    end
```

This seems to be the update feed used by the in-app updater.

Reply:
Cool! How did you manage to find that if you don't mind me asking? Network capture or something much more simple that I completely overlooked?

Reply:
I took a lazy approach and searched for "api.beeper.com" on GitHub, since update feeds are often used by various install scripts. If found [this line](https://github.com/gentoo/guru/blob/40982037cbf4fbee8ca4a9dab104d765c26e926a/net-im/beeper/beeper-4.0.570.ebuild#L18) and simply replaced the platform with `macos`.

Reply:
Now why didn't I think of that 😓 I did try searching their GitHub org for the new feed but was too fixated on the old todesktop.com url that I forgot to check the new api url...